crates/autopilot/src/domain/settlement/trade/mod.rs

```diff
@@ -0,0 +1,184 @@
+use {
+    super::{transaction, transaction::Prices},
+    crate::domain::{
+        self,
+        auction::{self, order},
+        eth,
+        fee,
+    },
+    bigdecimal::Zero,
+};
+
+mod math;
+
+/// Trade type evaluated in a context of an Auction.
+#[derive(Clone, Debug)]
+pub enum Trade {
+    /// A regular user order that exist in the associated Auction.
+    Fulfillment(Fulfillment),
+    // JIT trades are not part of the orderbook and are created by solvers at the time of
+    // settlement.
+    // Note that user orders can also be classified as JIT orders if they are settled outside of
+    // the Auction.
+    Jit(Jit),
+}
+
+impl Trade {
+    /// UID of the order that was settled in this trade.
+    pub fn uid(&self) -> &domain::OrderUid {
+        match self {
+            Self::Fulfillment(trade) => &trade.uid,
+            Self::Jit(trade) => &trade.uid,
+        }
+    }
+
+    /// Return JIT order if it's a JIT order.
+    pub fn as_jit(&self) -> Option<&Jit> {
```

Comment:
Interesting that the opposite `as_fulfillment` is not needed

Reply:
*so far 👍